src/gui/torrentcontenttreeview.cpp

```diff
@@ -106,6 +108,29 @@ void TorrentContentTreeView::keyPressEvent(QKeyEvent *event)
     }
 }
 
+void TorrentContentTreeView::mouseDoubleClickEvent(QMouseEvent *event)
+{
+    const QPersistentModelIndex index = indexAt(event->pos());
+    if (index.isValid())
```

Comment:
can you reorder it so we can save a level of indentation?
```c++
if (!index.isValid())
{
    QTreeView::mouseDoubleClickEvent(event);
    return;
}
// ...
```

Reply:
But then I have to call QTreeView::mouseDoubleClickEvent(event) twice. People have to follow two branches to understand the code.

Reply:
>can you reorder it so we can save a level of indentation?

`QTreeView::mouseDoubleClickEvent(event)` is called in all the cases, not only when index is invalid.

Reply:
>QTreeView::mouseDoubleClickEvent(event) is called in all the cases, not only when index is invalid.

Is it required to call it in all cases?
What if we don't call `QTreeView::mouseDoubleClickEvent(event)` in the case of clicking on the checkbox? (the event doesn't propagate further) Would that be wrong?

Reply:
The default implementation does other things, https://github.com/qbittorrent/qBittorrent/pull/14391#discussion_r575153764

Reply:
>The default implementation does other things, #14391 (comment)

Are they really required in the case of clicking on the checkbox? Did the single mouse click event (pressed & released) not triggered?

Reply:
Ideally we shouldn't look into its implementation. If we need to prevent "double click" on this checkbox we can do it in any way the interface allows us. However, there may be shortcomings in the implementation...
But why not try it first?

Reply:
> Are they really required in the case of double clicking on the checkbox?

I don't want to overwrite the default implementation, Qt never mentions that mouseDoubleClick is only used for this, overwriting will just create surprises for someone in the future when implementing a new feature in this view.